File: 2020/01/part-one.ts
import fs from 'fs-extra';

const rawInput = fs.readFileSync('./input.txt')
  .toString();
const input = rawInput.split(`
`)
  .map(el => Number(el));

for (let i = 0; i < input.length; i++) {
  for (let j = i + 1; j < input.length; j++) {
    const num1 = input[i];
    const num2 = input[j];
    if (num1 + num2 === 2020) {
      console.log(num1 * num2);
      break;
    }

  }
}

File: 2020/01/ruby/part-one.rb
file = File.open("./../input.txt")
input = file.read.split.map{|item| item.to_i}
file.close


for i in 0..input.length do
  for j in (i+1)..input.length do
    num1 = input[i]
    num2 = input[j]

    if num2.class == NilClass then
      next
    end

    sum = num1 + num2

    if(sum==2020)
      puts num1 * num2
    end
  end
end